services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -12,22 +13,59 @@ class ProjectService(projectRepository: ProjectRepository)(implicit executionCon
   def getProjectById(projectId: ProjectId): Future[Option[Project]] =
     projectRepository.getProjectById(projectId)
 
-  def addProject(request: ProjectAdditionRequest): Future[ProjectId] = {
+  def getProjectByName(namePattern: String, userId: UserId): Future[Option[Project]] = {
+    val result = projectRepository.getProjectByName(namePattern)
+    result.flatMap {
+      case Some(project) if project.ownerId == userId => result
+      case Some(_)                                    => Future.failed(new ProjectAccessDeniedException)
```

Comment:
Maybe we need to provide more context in this exception...

Reply:
I added more informative message of this exception

Reply:
done